jbmc/src/java_bytecode/java_bytecode_instrument.cpp

```diff
@@ -592,6 +593,30 @@ void java_bytecode_instrument_symbol(
   instrument(to_code(symbol.value));
 }
 
+/// Instruments the start function with an assertion that checks whether
+/// an exception has escaped the entry point
+/// \param symbol_table: global symbol table
+void java_bytecode_instrument_uncaught_exceptions(symbol_tablet &symbol_table)
```

Comment:
Is mutating the entry point a good plan? Might be better for the entry point construction to have a collection of `checks` that it appends after the return. Can't spot a concrete problem (so feel free to ignore) but I think strongly couples this code with the entry point generation (if the entry point code changes to wrap the whole thing in an if statement for whatever reason, the assert will be in the wrong place).

Reply:
In my opinion, the entire property instrumentation should be moved into a goto-pass.

Reply:
Be that as it may - is that likely to happen in the near future, and if not would decoupling be useful in the meantime?

Reply:
I've added some invariants to check that the entry point looks sane.